### PR TITLE
Handle null params in Spring Data Spanner

### LINKED ISSUE #2443: NullPointerException when passing the null value for nullable column
## Version
both spring-cloud-gcp-data-spanner-1.2.0.RELEASE and 1.2.1.RELEASE. Only I checked.

## What I did
I used `@Query` annotation and DML statement, then pass the null value for the nullable column.

## What was occurred
NullPointerException at SqlSpannerQuery.

## What I was expected
I want to input null value in the table if I passed null value to the method and the column is nullable.

## Situation the phenomena occurs
Assume that `lastName` is the nullable column at the singer table.

Define a method like below at the repository with extends of SpannerRepository
```java
@Query(dmlStatement = true, value = "INSERT INTO singer (singer_id, birth_date, first_name, last_name) VALUES (@singerId, @birthDay, @firstName, @lastName)")
public void insert(Integer singerId, Date birthDay, String firstName, String lastName);
```
Then use it like below
```java
xxxRepository.insert(3, Date.valueOf(LocalDate.of(1994, 12, 27)), "first_name", null);
```

### PR DESCRIPTION
**Pull request by dmitry-s**
fixes #2443

### CODE REVIEW

File: spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/repository/it/SpannerRepositoryInsertIntegrationTests.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		}
+		this.singerRepository.deleteAll();
+	}
+
+	@After
+	public void clearData() {
+		this.singerRepository.deleteAll();
+	}
+
+		@Test
```
Comment (dzou) [inline spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/repository/it/SpannerRepositoryInsertIntegrationTests.java:73]:
```suggestion
	@Test
```

File: spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java [hunk 2]
```diff
@@ -361,7 +361,8 @@ public static Statement buildStatementFromSqlWithArgs(String sql, List<String> t
 	private static void bindParameter(ValueBinder<Statement.Builder> bind,
 			Function<Object, Struct> paramStructConvertFunc, SpannerCustomConverter spannerCustomConverter,
 			Object originalParam, Parameter paramMetadata) {
-		if (ConversionUtils.isIterableNonByteArrayType(originalParam.getClass())) {
+		Class propType = originalParam != null ? originalParam.getClass() : paramMetadata.getType();
```
Comment (dzou) [inline spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java:364]: Can you add a comment here about how you need to know the type of the parameter (even if you bind `null`) which is why the type information is relevant for null?

**Comment by dzou** [comment]
Gonna close my PR in favor of this one.

File: spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/repository/query/SqlSpannerQueryTests.java [hunk 3]
```diff
@@ -426,72 +365,18 @@ public void compoundNameConventionTest() throws NoSuchMethodException {
...
 		when(this.queryMethod.getMethod()).thenReturn(method);
-
+		Mockito.<Parameters>when(this.queryMethod.getParameters()).thenReturn(new DefaultParameters(method));
 		sqlSpannerQuery.execute(params);
 
 		verify(this.spannerTemplate, times(1)).executeQuery(any(), any());
 	}
 
 	@Test
-	public void multiplePageableSortTest() {
```
Comment (meltsufin) [inline spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/repository/query/SqlSpannerQueryTests.java:440]: What's replacing these 3 tests that you deleted?

**Comment by dzou** [comment]
LGTM - only thing left is I think the build is broken.

**Comment by s13o** [comment]
Finally!
What is a planned release time?

**Comment by dmitry-s** [comment]
@s13o 
The next Spring Cloud release is scheduled for next week, but it can be delayed.

**Pull request merged**